src/main/java/io/mosip/digitalcard/service/impl/DigitalCardServiceImpl.java

```diff
@@ -102,35 +100,37 @@ public class DigitalCardServiceImpl implements DigitalCardService {
     @Value("${mosip.template-language}")
     private String templateLang;
 
-
     Logger logger = DigitalCardRepoLogger.getLogger(DigitalCardController.class);
 
     public void generateDigitalCard(String credential, String credentialType,String dataShareUrl,String eventId,String transactionId) {
-        boolean isGenerated = false;
         String decryptedCredential=null;
         String password=null;
+        String rid=null;
         try {
             if (dataShareUrl != null) {
                 credential = restClient.getForObject(dataShareUrl, String.class);
             }
             decryptedCredential = encryptionUtil.decryptData(credential);
-            JSONObject jsonObject = new org.json.JSONObject(decryptedCredential);
+            JSONObject jsonObject = new JSONObject(decryptedCredential);
             JSONObject decryptedCredentialJson = jsonObject.getJSONObject("credentialSubject");
+            rid=getRid(decryptedCredentialJson.get("id"));
             if (verifyCredentialsFlag){
                 logger.info("Configured received credentials to be verified. Flag {}", verifyCredentialsFlag);
                 boolean verified =credentialsVerifier.verifyCredentials(decryptedCredential);
                 if (!verified) {
                     logger.error("Received Credentials failed in verifiable credential verify method. So, digital card is not getting generated." +
                             " Id: {}, Transaction Id: {}",eventId, transactionId);
+                    digitalCardTransactionRepository.updateErrorTransactionDetails(rid,"ERROR","VC verification is failed.",LocalDateTime.now(),Utility.getUser());
```

Comment:
Instead of having direct error message like "VC verification is failed", can we rather have a error code configured in DigitalCardServiceErrorCodes and use both error code and error message concat string for the status comment ?

Reply:
taken care